### ISSUE TITLE
putup is running redundant git commands

### ISSUE THREAD

**Issue by julie777** [pyscaffold/pyscaffold#600]
## Description of your problem

**Please provide a minimal, self-contained, and reproducible example.**
Just run with --very-verbose.

**Please provide the full traceback using the `--very-verbose` flag.**
```python
    activate  pyscaffold.extensions.venv
    activate  pyscaffoldext.markdown.extension
      invoke  pyscaffold.actions:get_default_options
         run    git.exe --version
         run    git.exe --version
         run    git.exe config --get user.name
         run    git.exe config --get user.email
         run    git.exe config --get user.name
         run    git.exe config --get user.email
      invoke  pyscaffold.actions:verify_options_consistency
```

**Please provide any additional information below.**

## Versions and main components

* PyScaffold Version: 4.1.4
* Python Version: 3.10.2
* Operating system: Window 10/git bash
* How did you install PyScaffold: pipx

**Comment by julie777** [comment]
I haven't had time to really look into it. 

My impression from what I remember in the code is that we don't have a well defined initialization phase, so some parts of pysk just check preconditions before they run something, which doesn't matter for most things, but for external commands runs them multiple times.  (I hope the abbreviation is okay. I get tired of typing it out.)

**Comment by FlorianWilhelm** [comment]
Good point! That looks like a bug I guess, no reason to call it two times.
